be/src/exprs/string_functions.cpp

```diff
@@ -1226,7 +1235,7 @@ static inline ColumnPtr pad_utf8_const(Columns const& columns, BinaryColumn* src
 template <PadType pad_type>
 static inline ColumnPtr pad_const_not_null(const Columns& columns, BinaryColumn* src, const PadState* pad_state) {
     auto len = ColumnHelper::get_const_value<TYPE_INT>(columns[1]);
-    auto fill = ColumnHelper::get_const_value<TYPE_VARCHAR>(columns[2]);
+    auto fill = pad_state->fill;
```

Comment:
Not change here?
`pad_const_not_null ` is called only when `state->is_const`, not `state->fill_is_const`

```
template <PadType pad_type>
static ColumnPtr pad(FunctionContext* context, const Columns& columns) {
    RETURN_IF_COLUMNS_ONLY_NULL(columns);
    auto state = (PadState*)context->get_function_state(FunctionContext::FRAGMENT_LOCAL);
    if (state != nullptr) {
        if (state->is_const) {
            return pad_const<pad_type>(columns, state);
        } else if (state->fill_is_const) {
            return pad_not_const_check_ascii<true, pad_type>(columns, state);
        } else {
            return pad_not_const_check_ascii<false, pad_type>(columns, nullptr);
        }
    } else {
        return pad_not_const_check_ascii<false, pad_type>(columns, nullptr);
    }
}
```